tensorboard/program.py

```diff
@@ -218,6 +225,40 @@ def main(self, ignored_argv=('',)):
       sys.stderr.flush()
       return -1
 
+  def _register_info(self, server):
+    server_url = urllib.parse.urlparse(server.get_url())
+    info = manager.TensorboardInfo(
+        version=version.VERSION,
+        start_time=int(time.time()),
+        port=server_url.port,
+        pid=os.getpid(),
+        path_prefix=self.flags.path_prefix,
+        logdir=self.flags.logdir,
+        db=self.flags.db,
+        cache_key=manager.cache_key(
+            working_directory=os.getcwd(),
+            arguments=sys.argv[1:],
+        ),
+    )
+
+    @atexit.register
+    def clean_up():
+      manager.remove_info_file()
+
+    def handle_sigterm(signal_number, stack_frame):
+      del signal_number  # unused
+      del stack_frame  # unused
+      print("TensorBoard caught SIGTERM; exiting.")
+      # In case we get SIGTERMed again while running atexit handlers,
+      # take the hint and actually die.
+      signal.signal(signal.SIGTERM, signal.SIG_DFL)
```

Comment:
Slightly better I think to fetch the existing signal handler first (before installing this one) and call that instead.  By default it's still SIG_DFL but then in case some other part of the python process registers a signal handler first, we won't clobber it.

Reply:
Yes, true. I’d considered doing this but thought that it required a race
condition, but thinking about it again I don’t see why that has to be
the case. So, will change. Thanks.